# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# T2 SDE: package/.../xanim/devfs.patch
# Copyright (C) 2004 - 2005 The T2 SDE Project
# Copyright (C) 1998 - 2003 ROCK Linux Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---

--- ./docs/Rev_History.doc.orig	Wed Mar 21 14:33:32 2001
+++ ./docs/Rev_History.doc	Wed Mar 21 14:33:32 2001
@@ -244,7 +244,7 @@
                 This caused potential core dumps.
 
 	+ HPDEV:  setting sampling rate occasionally failed because 
-	          /dev/audio reported it was busy. Now loop on that.
+	          /dev/sound/audio reported it was busy. Now loop on that.
 
 -------------------------------------------------------------------------
 2.70.5 (04Apr96)
--- ./xa_audio.c.orig	Wed Mar 21 14:33:25 2001
+++ ./xa_audio.c	Wed Mar 21 14:33:26 2001
@@ -554,7 +554,7 @@
 
 
 /********** Sparc_Audio_Init **********************
- * Open /dev/audio and /dev/audioctl for Sparc's.
+ * Open /dev/sound/audio and /dev/sound/audioctl for Sparc's.
  *
  *****/
 void Sparc_Audio_Init()
@@ -566,7 +566,7 @@
 #endif
   DEBUG_LEVEL2 fprintf(stderr,"Sparc_Audio_Init\n");
   if (xa_audio_present != XA_AUDIO_UNK) return;
-  devAudio = open("/dev/audio", O_WRONLY | O_NDELAY);
+  devAudio = open("/dev/sound/audio", O_WRONLY | O_NDELAY);
   if (devAudio == -1)
   {
     if (errno == EBUSY) fprintf(stderr,"Audio_Init: Audio device is busy. - ");
@@ -665,7 +665,7 @@
 }
 
 /********** Sparc_Audio_Kill **********************
- * Close /dev/audio and /dev/audioctl.
+ * Close /dev/sound/audio and /dev/sound/audioctl.
  *
  *****/
 void Sparc_Audio_Kill()
@@ -1410,7 +1410,7 @@
 
 
 /********** EWS_Audio_Init **********************
- * Open /dev/audio/audio on EWS.
+ * Open /dev/sound/audio/audio on EWS.
  *
  *****/
 void EWS_Audio_Init()
@@ -1427,7 +1427,7 @@
  * It doesn't support FLUSH, or any of the other funky Sparc ioctls
  */
   if (xa_audio_present != XA_AUDIO_UNK) return;
-  devAudio = open("/dev/audio/audio", O_WRONLY | O_NDELAY);
+  devAudio = open("/dev/sound/audio/audio", O_WRONLY | O_NDELAY);
   if (devAudio == -1)
   {
     if (errno == EBUSY) fprintf(stderr,"Audio_Init: Audio device is busy. - ");
@@ -1462,7 +1462,7 @@
 }
 
 /********** EWS_Audio_Kill **********************
- * Close /dev/audio/audio
+ * Close /dev/sound/audio/audio
  *
  *****/
 void EWS_Audio_Kill()
@@ -2138,7 +2138,7 @@
   devMixer = open(_FILE_MIXER,  O_RDONLY | O_NDELAY, 0);
   /* Mixer only present on SB Pro's and above */
   /* if not present then it's set to -1 and ignored later */
-  /* THOUGHT: what about doing mixer ioctl to the /dev/dsp device??? */
+  /* THOUGHT: what about doing mixer ioctl to the /dev/sound/dsp device??? */
   if (devMixer < 0) devMixer = devAudio;
    /* determine what volume settings exist */
   { int devices;
@@ -3108,14 +3108,14 @@
 }
 
 /********** HPDEV_Audio_Init ***************************************************
- * open /dev/audio and init
+ * open /dev/sound/audio and init
  *****/
 void HPDEV_Audio_Init()
 {
   DEBUG_LEVEL2 fprintf(stderr,"HPDEV_Audio_Init\n");
   if (xa_audio_present != XA_AUDIO_UNK) return;
 
-  if ((devAudio = open ("/dev/audio", O_WRONLY | O_NDELAY, 0)) < 0)
+  if ((devAudio = open ("/dev/sound/audio", O_WRONLY | O_NDELAY, 0)) < 0)
   {
     if (errno == EBUSY) fprintf(stderr,"Audio_Init: Audio device is busy. - ");
     else fprintf(stderr,"Audio_Init: Error opening audio device. - ");
@@ -3127,10 +3127,10 @@
   fcntl(devAudio,F_SETFL,O_NDELAY);
 
 
-  /* Get description of /dev/audio: */
+  /* Get description of /dev/sound/audio: */
   if (ioctl (devAudio, AUDIO_DESCRIBE, &audioDesc))
   {
-    perror ("ioctl AUDIO_DESCRIBE on /dev/audio");
+    perror ("ioctl AUDIO_DESCRIBE on /dev/sound/audio");
     xa_audio_present = XA_AUDIO_ERR;
     return;
   }
@@ -3143,20 +3143,20 @@
 
   if (ioctl (devAudio, AUDIO_SET_DATA_FORMAT, AUDIO_FORMAT_LINEAR16BIT))
   {
-    perror ("ioctl AUDIO_SET_DATA_FORMAT on /dev/audio");
+    perror ("ioctl AUDIO_SET_DATA_FORMAT on /dev/sound/audio");
     xa_audio_present = XA_AUDIO_ERR;
     return;
   }
   if (ioctl (devAudio, AUDIO_SET_CHANNELS, 1))
   {
-    perror ("ioctl AUDIO_SET_CHANNELS on /dev/audio");
+    perror ("ioctl AUDIO_SET_CHANNELS on /dev/sound/audio");
     xa_audio_present = XA_AUDIO_ERR;
     return;
   }
 
   /* set device internal buffer size: */
   if (ioctl (devAudio, AUDIO_SET_TXBUFSIZE, 16 * 1024))
-    perror ("ioctl AUDIO_SET_TXBUFSIZE on /dev/audio");
+    perror ("ioctl AUDIO_SET_TXBUFSIZE on /dev/sound/audio");
 
   xa_interval_id = 0;
   xa_audio_present = XA_AUDIO_OK;
@@ -3188,7 +3188,7 @@
   xa_audio_status = XA_AUDIO_STOPPED;
 
   if (ioctl (devAudio, AUDIO_DRAIN, 0))
-    perror ("ioctl AUDIO_DRAIN on /dev/audio");
+    perror ("ioctl AUDIO_DRAIN on /dev/sound/audio");
 
   xa_time_audio = -1;
   xa_audio_flushed = 0;
@@ -3210,7 +3210,7 @@
     { 
       do { ret = ioctl(devAudio, AUDIO_SET_SAMPLE_RATE, xa_snd_cur->hfreq);
          } while( (ret < 0) && (errno == EBUSY) );
-      if (ret < 0) perror ("ioctl AUDIO_SET_SAMPLE_RATE on /dev/audio");
+      if (ret < 0) perror ("ioctl AUDIO_SET_SAMPLE_RATE on /dev/sound/audio");
       xa_audio_hard_freq = xa_snd_cur->hfreq;
     }
 
@@ -3273,7 +3273,7 @@
 
   if (ioctl (devAudio, AUDIO_GET_OUTPUT, &output_channel))
   {
-    perror ("ioctl AUDIO_GET_OUTPUT on /dev/audio");
+    perror ("ioctl AUDIO_GET_OUTPUT on /dev/sound/audio");
     output_channel = 0;
   }
   switch (flag)
@@ -3286,7 +3286,7 @@
       output_channel ^= AUDIO_OUT_SPEAKER;
   }
   if (ioctl (devAudio, AUDIO_SET_OUTPUT, output_channel))
-    perror ("ioctl AUDIO_SET_OUTPUT on /dev/audio");
+    perror ("ioctl AUDIO_SET_OUTPUT on /dev/sound/audio");
 }
 
 /********** HPDEV_Headphone_Toggle **********************
@@ -3302,7 +3302,7 @@
 
   if (ioctl (devAudio, AUDIO_GET_OUTPUT, &output_channel))
   {
-    perror ("ioctl AUDIO_GET_OUTPUT on /dev/audio");
+    perror ("ioctl AUDIO_GET_OUTPUT on /dev/sound/audio");
     output_channel = 0;
   }
   switch (flag)
@@ -3315,7 +3315,7 @@
       output_channel ^= AUDIO_OUT_HEADPHONE;
   }
   if (ioctl (devAudio, AUDIO_SET_OUTPUT, output_channel))
-    perror ("ioctl AUDIO_SET_OUTPUT on /dev/audio");
+    perror ("ioctl AUDIO_SET_OUTPUT on /dev/sound/audio");
 }
 
 /********** HPDEV_LineOut_Toggle **********************
@@ -3331,7 +3331,7 @@
 
   if (ioctl (devAudio, AUDIO_GET_OUTPUT, &output_channel))
   {
-    perror ("ioctl AUDIO_GET_OUTPUT on /dev/audio");
+    perror ("ioctl AUDIO_GET_OUTPUT on /dev/sound/audio");
     output_channel = 0;
   }
   switch (flag)
@@ -3344,7 +3344,7 @@
       output_channel ^= AUDIO_OUT_LINE;
   }
   if (ioctl (devAudio, AUDIO_SET_OUTPUT, output_channel))
-    perror ("ioctl AUDIO_SET_OUTPUT on /dev/audio");
+    perror ("ioctl AUDIO_SET_OUTPUT on /dev/sound/audio");
 }
 
 /********** HPDEV_Adjust_Volume **********************
@@ -3361,19 +3361,19 @@
 
   if (ioctl (devAudio, AUDIO_DESCRIBE, &description))
   {
-    perror ("ioctl AUDIO_DESCRIBE on /dev/audio");
+    perror ("ioctl AUDIO_DESCRIBE on /dev/sound/audio");
     return;
   }
   if (ioctl (devAudio, AUDIO_GET_GAINS, &gains))
   {
-    perror ("ioctl AUDIO_GET_GAINS on /dev/audio");
+    perror ("ioctl AUDIO_GET_GAINS on /dev/sound/audio");
     return;
   }
   gains.transmit_gain = (int)((float)description.min_transmit_gain +
 			      (float)(description.max_transmit_gain
 				      - description.min_transmit_gain) * floatvolume);
   if (ioctl (devAudio, AUDIO_SET_GAINS, &gains))
-    perror ("ioctl AUDIO_SET_GAINS on /dev/audio");
+    perror ("ioctl AUDIO_SET_GAINS on /dev/sound/audio");
 }
 
 #endif
@@ -4637,7 +4637,7 @@
 
 
 /********** NetBSD_Audio_Init **********************
- * Open /dev/audio and NetBSD.
+ * Open /dev/sound/audio and NetBSD.
  *
  *****/
 void NetBSD_Audio_Init()
@@ -4646,7 +4646,7 @@
   audio_info_t a_info;
   DEBUG_LEVEL2 fprintf(stderr,"NetBSD_Audio_Init\n");
   if (xa_audio_present != XA_AUDIO_UNK) return;
-  devAudio = open("/dev/audio", O_WRONLY | O_NDELAY);
+  devAudio = open("/dev/sound/audio", O_WRONLY | O_NDELAY);
   if (devAudio == -1)
   {
     if (errno == EBUSY) fprintf(stderr,"Audio_Init: Audio device is busy. - ");
@@ -4728,7 +4728,7 @@
 }
 
 /********** NetBSD_Audio_Kill **********************
- * Close /dev/audio.
+ * Close /dev/sound/audio.
  *
  *****/
 void NetBSD_Audio_Kill()
--- ./xa_audio.h.orig	Wed Mar 21 14:33:30 2001
+++ ./xa_audio.h	Wed Mar 21 14:33:30 2001
@@ -136,8 +136,8 @@
 #define _FILE_DSP "/dev/sb_dsp"
 #define _FILE_MIXER "/dev/sb_mixer"
 #else
-#define _FILE_DSP "/dev/dsp"
-#define _FILE_MIXER "/dev/mixer"
+#define _FILE_DSP "/dev/sound/dsp"
+#define _FILE_MIXER "/dev/sound/mixer"
 #ifdef __FreeBSD__
 #include <machine/soundcard.h>
 #else
--- ./xa_config.h.orig	Wed Mar 21 14:33:30 2001
+++ ./xa_config.h	Wed Mar 21 14:33:30 2001
@@ -292,6 +292,6 @@
 #ifdef XA_AIX_AUDIO
 #define DEFAULT_AUDIO_DEVICE_NAME   "/dev/paud0/1"
 #else
-#define DEFAULT_AUDIO_DEVICE_NAME   "/dev/audio"
+#define DEFAULT_AUDIO_DEVICE_NAME   "/dev/sound/audio"
 #endif
 
